routes/products.js

```diff
@@ -0,0 +1,130 @@
+const express = require("express");
+const db = require("../database");
+const router = express.Router();
+const app = express();
+
+router.get("/", (req, res) => {
+  db.raw(`SELECT * FROM products;`)
+    .then(results => {
+      res.json(results.rows);
+    })
+    .catch(err => {
+      res.json({ message: `Sorry, there was an error: ${err}` });
+    });
+});
+
+router.get("/:product_id", (req, res) => {
+  db.raw("SELECT * FROM products WHERE id = ?", [req.params.product_id])
+    .then(results => {
+      if (results.rows.length === 0) {
+        res.json({ message: "Product not found" });
+      } else {
+        res.json(results.rows[0]);
+      }
+    })
+    .catch(err => {
+      res.json({ message: `Sorry, there was an error: ${err}` });
+    });
+});
+
+router.post("/new", (req, res) => {
+  if (req.body.title === "" || req.body.description === "") {
+    res.json({ message: "Must POST all product fields" });
+  } else {
+    db.raw(
+      `INSERT INTO products (title, description, inventory, price) Values(?, ?, ?, ?) RETURNING *;`,
+      [req.body.title, req.body.description, req.body.inventory, req.body.price]
+    )
+      .then(results => {
+        res.json(results.rows[0]);
+      })
+      .catch(err => {
+        res.json({ message: "Must POST all product fields" });
+      });
+  }
+});
+
+router.put("/:product_id", (req, res) => {
+  req.body.title.trim();
+  req.body.description.trim();
+  req.body.inventory.trim();
+  req.body.price.trim();
+  let title;
+  let description;
+  let inventory;
+  let price;
+  db.raw("SELECT * FROM products where id = ?", [req.params.product_id])
```

Comment:
These promises are not chained. Also instead of checking each attribute individually, it's easier to just get the entire object from the DB then do your validations in code.